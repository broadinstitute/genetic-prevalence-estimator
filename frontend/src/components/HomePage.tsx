import { Box, Heading, Link, Text } from "@chakra-ui/react";
import { Redirect } from "react-router-dom";

import SignInButton from "./SignInButton";
import { authStore, useStore } from "../state";

const HomePage = () => {
  const { isSignedIn } = useStore(authStore);

  if (isSignedIn) {
    return <Redirect to="/variant-lists/" />;
  }

  return (
    <>
      <Heading as="h1" mb={4}>
        Genetic Prevalence Calculator
      </Heading>

      <Text mb={8}>
        This tool aims to estimate carrier frequency and genetic prevalence for
        recessive conditions using allele frequencies from population data in
        gnomAD. We provide support for creating variant lists from{" "}
        <Link href="https://www.ncbi.nlm.nih.gov/clinvar/" isExternal>
          ClinVar
        </Link>{" "}
        and{" "}
        <Link href="https://gnomad.broadinstitute.org/" isExternal>
          gnomAD
        </Link>
        , or allow users to build their own custom list using gnomAD IDs. Sign
        in with your Google account to get started.
      </Text>

      <Box mb={4}>
        <SignInButton />
      </Box>

      <details>
        <summary>Why do I need to sign in?</summary>
        <Box borderLeftWidth="1px" borderColor="gray.400" pl={4} mt={2}>
          In order to store your variant lists, and allow you to edit them over
          time we need to have you signed in. By having users sign in we are
          also able to allow sharing lists across users, enabling collaboration.
        </Box>
      </details>
    </>
  );
};

export default HomePage;
