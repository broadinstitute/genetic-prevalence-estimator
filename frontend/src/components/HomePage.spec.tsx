import { expect, test } from "@jest/globals";
import renderer from "react-test-renderer";
import { withDummyRouter } from "../../tests/__helpers__/router";

import HomePage from "./HomePage";

test("About Page has no unexpected changes", () => {
  const tree = renderer.create(withDummyRouter(<HomePage />));
  expect(tree).toMatchSnapshot();
});
