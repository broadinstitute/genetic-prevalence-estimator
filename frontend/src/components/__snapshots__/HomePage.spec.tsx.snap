// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page has no unexpected changes 1`] = `
Array [
  <h1
    className="chakra-heading css-1itv5e3"
  >
    GeniE, the Genetic Prevalence Estimator
  </h1>,
  <p
    className="chakra-text css-19idom"
  >
    The prevalence GeniE tool aims to estimate carrier frequency and genetic prevalence for recessive conditions using allele frequencies from population data in gnomAD. We provide support for creating variant lists from
     
    <a
      className="chakra-link css-0"
      href="https://www.ncbi.nlm.nih.gov/clinvar/"
      rel="noopener"
      target="_blank"
    >
      ClinVar
    </a>
     
    and
     
    <a
      className="chakra-link css-0"
      href="https://gnomad.broadinstitute.org/"
      rel="noopener"
      target="_blank"
    >
      gnomAD
    </a>
    , or allow users to build their own custom list using gnomAD IDs. Sign in with your Google account to get started.
  </p>,
  <div
    className="css-1itv5e3"
  >
    <div />
  </div>,
  <details>
    <summary>
      Why do I need to sign in?
    </summary>
    <div
      className="css-1wyzu5o"
    >
      In order to store your variant lists, and allow you to edit them over time we need to have you signed in. By having users sign in we are also able to allow sharing lists across users, enabling collaboration.
    </div>
  </details>,
]
`;
